enterprise/backup/src/test/java/org/neo4j/backup/OnlineBackupCommandConfigLoaderTest.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.backup;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import org.neo4j.causalclustering.core.CausalClusteringSettings;
+import org.neo4j.commandline.admin.CommandFailed;
+import org.neo4j.graphdb.config.InvalidSettingException;
+import org.neo4j.kernel.configuration.Config;
+
+import static java.lang.String.format;
+import static org.hamcrest.Matchers.any;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+public class OnlineBackupCommandConfigLoaderTest
+{
+    @Rule
+    public TemporaryFolder testDirectory = new TemporaryFolder();
```

Comment:
Why `TemporaryFolder` instead of `TestDirectory`?

Reply:
I think we really should use `TestDirectory` because that will put the data in `target/test-data`, which on our build agents is more likely to be mounted on a volume that can fit the data, than `TMPDIR` is.

Reply:
Addressed in next commit